File: app/src/main/java/com/chiranths/jobportal1/Adapters/ConstructorAdaptor.java
package com.chiranths.jobportal1.Adapters;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.chiranths.jobportal1.Activities.Construction.ConstructionDetailsActivity;
import com.chiranths.jobportal1.Model.ConstructionModel;
import com.chiranths.jobportal1.R;
import com.chiranths.jobportal1.Utilitys.AppConstants;
import com.chiranths.jobportal1.Utilitys.Utilitys;

import java.util.List;

public class ConstructorAdaptor extends RecyclerView.Adapter<ConstructorAdaptor.ViewHolder> {

    private List<ConstructionModel> productInfos;
    private Context context;
    private String number, name;
    Utilitys utilitys = new Utilitys();

    public ConstructorAdaptor(List<ConstructionModel> productInfos, Context context) {
        this.productInfos = productInfos;
        this.context = context;
        this.number = number;
        this.name = name;
    }

    @NonNull
    @Override
    public ConstructorAdaptor.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        context = parent.getContext();
        View listItem= layoutInflater.inflate(R.layout.activity_construction_adaptor, parent, false);
        ConstructorAdaptor.ViewHolder viewHolder = new ConstructorAdaptor.ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ConstructorAdaptor.ViewHolder holder, @SuppressLint("RecyclerView") int position) {

        ConstructionModel productInfo = productInfos.get(position);
        Glide.with(context)
                .load(productInfo.getImage())
                .into(holder.iv_vehicle_image);

        holder.travel_vehicle_name.setText(productInfo.getName());
        holder.tv_travel_category.setText(productInfo.getCategory());
        holder.tv_cost_km.setText(productInfo.getCost());
        holder.tv_vehicle_number.setText(productInfo.getExperience());

        holder.iv_const_call_bottom.setOnClickListener(view -> {
            utilitys.navigateCall(context,productInfo.getContactDetails(),productInfo.getName());
        });

        holder.iv_const_whatsapp_bottom.setOnClickListener(view -> {
            utilitys.navigateWhatsapp(context,productInfo.getContactDetails(),productInfo.getName());

        });

        holder.cv_card_const.setOnClickListener(view -> {
            Intent intent =new Intent(context, ConstructionDetailsActivity.class);
            intent.putExtra(AppConstants.pid,productInfo.getPid());
            context.startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return productInfos.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ImageView iv_vehicle_image,iv_const_whatsapp_bottom,iv_const_call_bottom;
        TextView travel_vehicle_name,tv_travel_category,tv_cost_km,tv_vehicle_number,tv_discription;
        CardView cv_card_const;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            travel_vehicle_name = itemView.findViewById(R.id.travel_vehicle_name);
            tv_travel_category = itemView.findViewById(R.id.tv_travel_category);
            tv_cost_km = itemView.findViewById(R.id.tv_cost_km);
            tv_vehicle_number = itemView.findViewById(R.id.tv_vehicle_number);
            iv_vehicle_image = itemView.findViewById(R.id.iv_vehicle_image);
            iv_const_whatsapp_bottom = itemView.findViewById(R.id.iv_const_whatsapp_bottom);
            iv_const_call_bottom = itemView.findViewById(R.id.iv_const_call_bottom);
            cv_card_const = itemView.findViewById(R.id.cv_card_const);
        }
    }

}

File: app/src/main/java/com/sbd/gbd/Adapters/ConstructorAdaptor.kt
package com.sbd.gbd.Adapters

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.google.firebase.database.FirebaseDatabase
import com.sbd.gbd.Activities.Construction.ConstructionDetailsActivity
import com.sbd.gbd.Model.ConstructionModel
import com.sbd.gbd.R
import com.sbd.gbd.Utilitys.AppConstants
import com.sbd.gbd.Utilitys.Utilitys

class ConstructorAdaptor(
    private val productInfos: ArrayList<ConstructionModel>,
    private var context: Context
) : RecyclerView.Adapter<ConstructorAdaptor.ViewHolder>() {
    var utilitys = Utilitys()


    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        context = parent.context
        val listItem =
            layoutInflater.inflate(R.layout.activity_construction_adaptor, parent, false)
        return ViewHolder(listItem)
    }

    override fun onBindViewHolder(holder: ViewHolder, @SuppressLint("RecyclerView") position: Int) {
        val productInfo = productInfos[position]
        Glide.with(context)
            .load(productInfo.image)
            .into(holder.iv_vehicle_image)
        holder.travel_vehicle_name.text = productInfo.name
        holder.tv_travel_category.text = productInfo.category
        holder.tv_cost_km.text = productInfo.cost
        holder.tv_vehicle_number.text = productInfo.experience
        holder.iv_const_call_bottom.setOnClickListener { view: View? ->
            utilitys.navigateCall(
                context, productInfo.contactDetails, productInfo.name
            )
        }

        if (productInfo.status.equals("1")) {
            holder.ll_approve.visibility = View.VISIBLE
            holder.ll_call?.visibility = View.GONE
        } else if (productInfo.status.equals("3")) {
            holder.ll_approve.visibility = View.VISIBLE
            holder.ll_call?.visibility = View.GONE
        } else {
            holder.ll_approve.visibility = View.GONE
            holder.ll_call?.visibility = View.VISIBLE
        }

        holder.ll_approve.setOnClickListener {
            productInfo.pid.let { it1 ->
                if (it1 != null) {
                    FirebaseDatabase.getInstance().reference.child(AppConstants.construction).child(it1)
                        .child(AppConstants.Status).setValue("2")
                }
            }
        }

        holder.iv_const_whatsapp_bottom.setOnClickListener { view: View? ->
            utilitys.navigateWhatsapp(
                context, productInfo.contactDetails, productInfo.name
            )
        }
        holder.cv_card_const.setOnClickListener { view: View? ->
            val intent = Intent(context, ConstructionDetailsActivity::class.java)
            intent.putExtra(AppConstants.pid, productInfo.pid)
            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return productInfos.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var iv_vehicle_image: ImageView
        var iv_const_whatsapp_bottom: ImageView
        var iv_const_call_bottom: ImageView
        var travel_vehicle_name: TextView
        var tv_travel_category: TextView
        var tv_cost_km: TextView
        var tv_vehicle_number: TextView
        var tv_discription: TextView? = null
        var cv_card_const: CardView
        var ll_approve: LinearLayout
        var ll_call: LinearLayout

        init {
            travel_vehicle_name = itemView.findViewById(R.id.travel_vehicle_name)
            tv_travel_category = itemView.findViewById(R.id.tv_travel_category)
            tv_cost_km = itemView.findViewById(R.id.tv_cost_km)
            tv_vehicle_number = itemView.findViewById(R.id.tv_vehicle_number)
            iv_vehicle_image = itemView.findViewById(R.id.iv_vehicle_image)
            iv_const_whatsapp_bottom = itemView.findViewById(R.id.iv_const_whatsapp_bottom)
            iv_const_call_bottom = itemView.findViewById(R.id.iv_const_call_bottom)
            cv_card_const = itemView.findViewById(R.id.cv_card_const)
            ll_approve = itemView.findViewById(R.id.ll_approve)
            ll_call = itemView.findViewById(R.id.ll_call)
        }
    }
}
